Add faction specific data, initial work

lstrzebinczyk opened the pull request:
I am submitting initial PR about inclusion of faction-specific data with you, in order to agree on the approach.

This PR does the following:

1. Expands the `Roster` type with a `faction` string
2. Expands parsing of the KT2021 roster to fetch the faction from it
3. Expands the `Roster` component from KT2021 with a `faction` prop
4. Passess the `faction`, as parsed from the Battlescribe roster, to the `Roster` component.
5. Adds `./data/veteranGuardsmen,ts` file, which exports an object containing veteran guardsmen specific strategic ploys, tactical ploys and tac ops
6. Adds `./data/index`, which exports a function. Function accepts the name of a faction, and returns faction speficic strategic ploys etc.
7. Adds `PloysColumn` component. At the time, it's nearly a copy-paste of the `PowerList`
8. In the `Roster` component fetch the faction specific data, and if it's there, render the strategic ploys and tactical ploys using the `PloysColumn` component, and some styling.

Like I mentioned, this is an initial version, so that I can show you how I am thinking to organise this and get initial feedback.
This is missing the following:
1. Tac Ops rendering
2. I would like to render the ploys as 2 parallel columns, as it is in the Octarius book. They need prices present as well.

Thoughts?

Floppy commented on src/components/KillTeam2021/data/index.tsx:
```diff
@@ -0,0 +1,12 @@
+
+const getData = (factionName: string) => {
+  switch (factionName) {
+    case "Veteran Guardsmen":
```
Ah, there's a `camelCase` method in `lodash`, which we're already using, so no need for a new dependency: https://lodash.com/docs/4.17.15#camelCase

lstrzebinczyk replied:
@Floppy Can you share more about how the solution with `camelCase` would work? I think I'm not seeing what you are seeing, because the only thing coming to my mind is doing dynamic import, but that would make the function async, and that would be tricky to deal with, I think?

lstrzebinczyk commented:
Thank you very much, I will apply your suggestions later today (hopefully). One thing I noticed additionally is that maybe it would be good to include `Veteran Guardsman Tactical Assets` ? 

But that poses a new issue: those are unique to the veteran guardsman. Similarily to grey knights powers. One way to accomodate for that would be this: Instead of storing object containing data, and use them to render faction specific content inside `Roster`, we implement a faction specific footer component, and just include that and render it at the end of the roster. It could contain ploys, tac ops, as well as all additional faction-specific information. It would require an implementation of all of that, but I envision a future in which all information required for playing are on the printed dataslate roster: )

Floppy commented:
Could work! I suspect they're in as reference data in the roster file, which is the same as the psychic powers and orders. We can create another ticket to remind us :)

lstrzebinczyk commented on src/components/KillTeam2021/data/index.tsx:
```diff
@@ -0,0 +1,12 @@
+import veteranGuardsmen from "./veteranGuardsmen"
```
Done.

lstrzebinczyk commented on src/components/KillTeam2021/data/index.tsx:
```diff
@@ -0,0 +1,12 @@
+import veteranGuardsmen from "./veteranGuardsmen"
+
+const getData = (factionName: string) => {
```
Renamed to `getFactionSpecificData` :+1:

lstrzebinczyk commented on src/components/KillTeam2021/data/veteranGuardsmen.ts:
```diff
@@ -0,0 +1,76 @@
+  id: 1,
+  name: 'Boots on the ground',
+  revealTime: 'You can reveal this Tac Op at the end of any Turning Point',
+  line1: `At the end of that Turning Point, if there are more friendly operatives than enemy operatives within ⬟ of your drop zone, and there are more friendly
```
Good thinking! Fixed.

lstrzebinczyk commented:
@Floppy I adjusted the code to your suggestions (except for 1 case which wasn't clear to me). I also found a bug and created a ticket about it and added the ploys and tac ops for kommando.

What's our next move? : )

Floppy commented:
Fantastic, I'll run it up here and then merge it in. Nice work :)
